optimum/onnxruntime/base.py

```diff
@@ -0,0 +1,397 @@
+#  Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Defines the base classes that are used to perform inference with ONNX Runtime of Transformers models."""
+
+from abc import abstractmethod
+from typing import Dict, List, Optional, Tuple
+
+import torch
+from transformers.modeling_outputs import BaseModelOutput, CausalLMOutputWithCrossAttentions, Seq2SeqLMOutput
+
+from onnxruntime import InferenceSession
+
+from ..utils import NormalizedConfigManager
+
+
+class ORTModelPart:
+    """
+    For multi-file ONNX models, such as encoder-decoder models, represents a part of the model.
+    It has its own `onnxruntime.InferenceSession`, and can perform a forward pass.
+    """
+
+    def __init__(
+        self,
+        session: InferenceSession,
+        parent_model: "ORTModel",
```

Comment:
I'm not a huge fan of passing a parent class to its children; it feels like it can be bug prone. Couldn't we simply pass the relevant attributes, and possibly update them when needed (e.g. device)

Also, not sure if the parent model will actually have a `.config` attribute, e.g. see https://huggingface.co/stabilityai/stable-diffusion-2-1/tree/main . But we can leave it to an other PR to make it more general.